packages/xrpl/src/models/transactions/common.ts

```diff
@@ -400,15 +400,19 @@ export function validateBaseTransaction(common: Record<string, unknown>): void {
  *
  * @param amount - An Amount to parse for its value.
  * @returns The parsed amount value, or NaN if the amount count not be parsed.
+ * @throws ValidationError, if the input Amount is invalid
+ * @throws SyntaxError, if Amount cannot be parsed by BigInt constructor
  */
-export function parseAmountValue(amount: unknown): number {
+export function parseAmountValue(amount: unknown): bigint {
```

Comment:
Couldn't the amount be a floating point number? In that case, it would be `number` instead of `bigint`. `bigint` doesn't support floating point numbers.

Reply:
Thanks for pointing this out. I missed the fact that `bigint` cannot handle floats.

I wrote this PR with the perspective of `AssetPrice` field in OracleSet transaction which only accepts `std::uint_64` values.

However, you are correct. We cannot make this assumption for the rest of `Amount` fields. Do you have any suggestion for javascript libraries which accept floating-point numbers and also are compatible with rippled's `AMOUNT` internal-type field?

Reply:
You can just make the return type a union: `number | bigint`